server/src/test/java/org/opensearch/ingest/IngestServiceTests.java

```diff
@@ -135,65 +135,33 @@ public Map<String, Processor.Factory> getProcessors(Processor.Parameters paramet
     };
 
     private ThreadPool threadPool;
-    private BulkRequest mockBulkRequest;
+    private Settings settings = Settings.EMPTY;
 
     @Before
     public void setup() {
         threadPool = mock(ThreadPool.class);
         ExecutorService executorService = OpenSearchExecutors.newDirectExecutorService();
         when(threadPool.generic()).thenReturn(executorService);
         when(threadPool.executor(anyString())).thenReturn(executorService);
-        mockBulkRequest = mock(BulkRequest.class);
-        lenient().when(mockBulkRequest.batchSize()).thenReturn(1);
```

Comment:
Apologies in advance to reviews, there are a couple small refactorings here that weren't strictly necessary:

- This mock is needed at all. I replaced all usages with `new BulkRequest()`.
- I replaced the custom `mapOf` functions with the JDK-provided ones